### PR TITLE
Admin product attribute set save controller refactor

### PR DESCRIPTION
**Pull request by AnshuMishra17**
### Description
Removed the direct use of object Manager from class Save in namespace Magento\Catalog\Controller\Adminhtml\Product\Set using controller dependency injection

### Manual testing scenarios
1. Added a new attribute set from Admin: Stores -> Attribute Set -> Add Attribute Set
2. Updated the attribute set in Stores -> Attribute Set

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=15850) <br/>All committers have signed the CLA.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Set/Save.php [hunk 1]
```diff
@@ -17,22 +17,46 @@ class Save extends \Magento\Catalog\Controller\Adminhtml\Product\Set
...
      */
     public function __construct(
         \Magento\Backend\App\Action\Context $context,
         \Magento\Framework\Registry $coreRegistry,
         \Magento\Framework\View\LayoutFactory $layoutFactory,
-        \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory
+        \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
+        \Magento\Eav\Model\Entity\Attribute\SetFactory $attributeSetFactory = null,
+        \Magento\Framework\Filter\FilterManagerFactory $filterManagerFactory = null,
+        \Magento\Framework\Json\Helper\DataFactory $jsonHelperFactory = null
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/Set/Save.php:52]: We don't have any places where this class was created via factory.

Example: https://github.com/magento/magento2/blob/9c14af7a97bcf338662de365d2613c7ca9dbf8c6/app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php#L44

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Set/Save.php [hunk 2]
```diff
@@ -17,22 +17,46 @@ class Save extends \Magento\Catalog\Controller\Adminhtml\Product\Set
...
+     * @param \Magento\Framework\Json\Helper\DataFactory $jsonHelperFactory
      */
     public function __construct(
         \Magento\Backend\App\Action\Context $context,
         \Magento\Framework\Registry $coreRegistry,
         \Magento\Framework\View\LayoutFactory $layoutFactory,
-        \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory
+        \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
+        \Magento\Eav\Model\Entity\Attribute\SetFactory $attributeSetFactory = null,
+        \Magento\Framework\Filter\FilterManagerFactory $filterManagerFactory = null,
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/Set/Save.php:51]: We don't have any places where this class was created via factory.


Example:
https://github.com/magento/magento2/blob/268dce1cded190e7083d8ec9c864500dec254147/app/code/Magento/Store/Model/Address/Renderer.php#L48

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Could you squash all your changes into one commit and force push?

**Comment by AnshuMishra17** [comment]
@ihor-sviziev  I don't know how to squash all commits and force push, I have tried couple of times but I was unsuccessful. It would be helpful if you can share the the commands for it and little explanation.
For now, I am closing this PR and will generate a new one.

**Comment by ihor-sviziev** [comment]
Hi @AnshuMishra17,
Here is example: https://github.com/magento/magento2/pull/15337#issuecomment-394604296 but you need to use 2.2-develop branch and your changes.

Actually it's better to continue this PR than creating new one.

**UPD:**
OK. I see you already created new PR https://github.com/magento/magento2/pull/15876. I'll process it

**Pull request closed without merging**